### ISSUE TITLE
Unable to open the file that the filename contains '%'

### ISSUE THREAD

**Issue by stevenlee7189** [conwnet/github1s#439]
github1s returns 400 bad request if the filename contains '%'

For example:
https://github1s.com/AspeedTech-BMC/openbmc/blob/aspeed-master/meta-aspeed-sdk/meta-ast2600-sdk/recipes-bsp/u-boot/u-boot-aspeed-sdk_2019.04%.bbappend

**Comment by Siddhant-K-code** [comment]
The URL you provided doesn't seem to work on GitHub either. I tried it removing the `1s`. I got the following page:

![image](https://user-images.githubusercontent.com/55068936/187900793-50fbb755-b19a-476a-87d3-29fe663f1720.png)


But, if you try to open the same from GitHub UI (through repository), you will notice that GitHub automatically adds a number called `25` after `%`. This is the URL that I got from GitHub: `https://github.com/AspeedTech-BMC/openbmc/blob/2214c648537a275b174f01be678052456879b138/meta-aspeed-sdk/meta-ast2600-sdk/recipes-bsp/u-boot/u-boot-aspeed-sdk_2019.04%25.bbappend` while opening the file `openbmc/meta-aspeed-sdk/meta-ast2600-sdk/recipes-bsp/u-boot/u-boot-aspeed-sdk_2019.04%.bbappend`


I think, our middleware are unable to redirect to the same & searching for the exact file, that's why it is returning 404.

WDYT, any possible way to fix this? @conwnet
